h2o-core/src/main/java/hex/ModelMetrics.java

```diff
@@ -135,7 +136,20 @@ public static double getMetricFromModelMetric(ModelMetrics mm, String criterion)
     Method method = null;
     Object obj = null;
     criterion = criterion.toLowerCase();
-    ConfusionMatrix cm = mm.cm();
+    
+    // Constructing confusion matrix based on criterion
+    ConfusionMatrix cm;
+    AUC2.ThresholdCriterion criterionAsEnum = null;
+    try {
```

Comment:
What is the purpose of the try-catch?

Reply:
@michalkurka the reason is that criterion might not be in enum. For example by default `logloss` will be passed into this method and it is not present in AUC2.ThresholdCriterion enum. There is no `exist` method in enums. I can rewrite it with searching through all values  approach. Or what was exactly your concern about?

Reply:
Catching an exception without handling in any way is an anti-pattern. From your description it doesn't seem like this is necessary, please fix.

Reply:
@michalkurka I know that it is anti-pattern. It is just being used all over the method(and also whole code base) at lines 157 and 166. We should at least log something in the catch blocks, otherwise it might be a blackhole. I will fix